lib/utils.js

```diff
@@ -0,0 +1,46 @@
+var Q = require('q');
+var es = require('.././config/es');
+
+var log = console.log;
+
+exports = module.exports = {
+  waitForElastic: function (waitCnt) {
+    var waitCntDef = 50;
+    waitCnt = waitCnt || 50;  // 40 requests by default
+
+    var deferred = Q.defer();
+
+    (function pinger(done) {
+      if(waitCnt != waitCntDef) log('requests left', waitCnt);
+
+      if(waitCnt-- == 0) {
+        deferred.reject(new Error('Max requests exceeded, elastic is dead.'));
```

Comment:
You use `deferred.reject` here, but also use `done` for callback. There is something strange code :-)